src/main.rs

```diff
@@ -544,36 +574,48 @@ async fn main() -> Result<()> {
     let mut cases: Vec<_> = ctx.retrieve_case_infos().await?.into_iter().collect();
     let num_cases = cases.len();
     let one_case = num_cases == 1;
+    let output = &mut ctx.ctx_mut().output;
 
     if one_case && original_output.is_none() {
         // We need to update the output name, now that we know the title.
         let mut name = cases.first().unwrap().filename();
         if one_file {
             name += ".html";
         }
-        ctx.output = PathBuf::from(name);
+        *output = PathBuf::from(name);
     } else if one_case
         && one_file
-        && ctx
-            .output
+        && output
             .extension()
             .is_none_or(|x| x.to_ascii_lowercase() != "html")
     {
-        if ctx.output.is_dir() {
-            ctx.output = ctx.output.join(cases.first().unwrap().filename());
+        if output.is_dir() {
+            *output = output.join(cases.first().unwrap().filename());
         }
-        ctx.output.set_extension("html");
+        output.set_extension("html");
+    } else if !one_case
+        && original_output.is_none()
+        && cases
+            .iter()
+            .map(|x| &x.case_information.sequence)
+            .all_equal_value()
+            .is_ok_and(Option::is_some)
+    {
+        // All downloaded cases are part of a sequence.
+        let sequence = cases
+            .first()
+            .as_ref()
+            .unwrap()
+            .case_information
+            .sequence
+            .as_ref()
+            .unwrap();
+        *output = PathBuf::from(&sequence.title);
     }
+    let output = &ctx.ctx().output;
 
     // Empty directories are fine if they exist already.
-    if ctx.output.exists()
-        && ctx
-            .output
-            .read_dir()
-            .ok()
-            .and_then(|mut x| x.next())
-            .is_some()
-    {
+    if output.exists() && output.read_dir().ok().and_then(|mut x| x.next()).is_some() {
         if ctx.ctx().args.remove_existing {
```

Comment:
In my intuition, I would not necessarily expect `--remove-existing` to delete _all_ files in the directory if it already exists.

It would be really bad if you'd call this with e.g. `$HOME`, expecting it to only delete the files related to `aaoffline`, and then lose everything in your home directory (although you could also argue that one should RTFM before using it).

At the very least, I'd add a confirmation prompt to make sure the user knows that the entire directory is about to be deleted.

Reply:
Hm, yes, you're probably right. I would've said "a parameter called `--remove-existing` should be explicit enough to not warrant a confirmation prompt", but I fell into this trap myself once and accidentally deleted a bunch of progress I made on `aaoffline` because I specified `-o .` together with `--remove-existing`… 

A better solution would probably be to just rename this parameter to `--overwrite-existing` and then overwrite any existing files instead of deleting the whole directory first. I'll try to implement this later.

Reply:
This is now implemented in d7476f52da08fbd622275f7b42686d09f4b540b6.